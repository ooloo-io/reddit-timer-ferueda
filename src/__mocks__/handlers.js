import { rest } from 'msw';
import response1 from './response-1.json';
import response2 from './response-2.json';
import response3 from './response-3.json';
import response4 from './response-4.json';
import response5 from './response-5.json';

const responseMap = {
  initial: response1,
  t3_f807cu: response2,
  t3_f1r3pr: response3,
  t3_i51dz8: response4,
  t3_eifs0j: response5,
};

const getJSONResponse = (req) => {
  const after = req.url.searchParams.get('after');
  return responseMap[after || 'initial'];
};

const handlers = [
  rest.get('https://www.reddit.com/r/lessthan500/top.json?t=year&limit=100', (req, res, ctx) => {
    const posts = getJSONResponse(req);

    if (req.url.searchParams.get('after') === 't3_f1r3pr') {
      posts.data.after = null;
      posts.data.children = posts.data.children.slice(0, 70);
    }

    return res(ctx.status(200), ctx.json(posts));
  }),
  rest.get('https://www.reddit.com/r/failed/top.json?t=year&limit=100', (req, res, ctx) => res(ctx.status(500), ctx.json({ errorMessage: 'Internal server error' }))),
  rest.get('https://www.reddit.com/*', (req, res, ctx) => res(ctx.status(200), ctx.json(getJSONResponse(req)))),
];

export default handlers;
