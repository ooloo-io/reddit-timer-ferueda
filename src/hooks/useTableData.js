import React from 'react';
import { daysArray } from '../constants/daysAndTime';
import { getDayAndTimeFromPosts, getFrequencyByDayAndTime } from '../utils/helpers';

const useTableData = (posts) => {
  const postsData = React.useMemo(() => {
    const frequency = getFrequencyByDayAndTime(getDayAndTimeFromPosts(posts));

    return daysArray.map((day) => frequency[day]);
  }, [posts]);

  const tableColumns = React.useMemo(() => {
    const columnsArray = [];

    for (let i = 0; i <= 23; i += 1) {
      columnsArray.push({ accessor: i.toString() });
    }

    return columnsArray;
  }, []);

  return {
    postsData,
    tableColumns,
  };
};

export default useTableData;
