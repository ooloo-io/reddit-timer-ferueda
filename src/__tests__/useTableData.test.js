import { renderHook } from '@testing-library/react-hooks';
import useTableData from '../hooks/useTableData';
import { posts, formatedData, emptyData } from '../__mocks__/useTableData';

describe('Hook - useTableData', () => {
  test('transforms data from 500 posts', () => {
    const { result } = renderHook(() => useTableData(posts));

    expect(result.current.postsData).toEqual(formatedData);
    expect(result.current.tableColumns).toEqual(emptyData.columns);
    expect(result.current.tableColumns).toMatchSnapshot();
    expect(result.current.postsData).toMatchSnapshot();
  });
});
