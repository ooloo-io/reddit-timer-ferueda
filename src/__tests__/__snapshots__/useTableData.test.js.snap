// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hook - useTableData transforms data from 500 posts 1`] = `
Array [
  Object {
    "accessor": "0",
  },
  Object {
    "accessor": "1",
  },
  Object {
    "accessor": "2",
  },
  Object {
    "accessor": "3",
  },
  Object {
    "accessor": "4",
  },
  Object {
    "accessor": "5",
  },
  Object {
    "accessor": "6",
  },
  Object {
    "accessor": "7",
  },
  Object {
    "accessor": "8",
  },
  Object {
    "accessor": "9",
  },
  Object {
    "accessor": "10",
  },
  Object {
    "accessor": "11",
  },
  Object {
    "accessor": "12",
  },
  Object {
    "accessor": "13",
  },
  Object {
    "accessor": "14",
  },
  Object {
    "accessor": "15",
  },
  Object {
    "accessor": "16",
  },
  Object {
    "accessor": "17",
  },
  Object {
    "accessor": "18",
  },
  Object {
    "accessor": "19",
  },
  Object {
    "accessor": "20",
  },
  Object {
    "accessor": "21",
  },
  Object {
    "accessor": "22",
  },
  Object {
    "accessor": "23",
  },
]
`;

exports[`Hook - useTableData transforms data from 500 posts 2`] = `
Array [
  Object {
    "0": 2,
    "1": 1,
    "10": 6,
    "11": 3,
    "12": 6,
    "13": 4,
    "14": 3,
    "15": 5,
    "16": 3,
    "17": 1,
    "18": 3,
    "19": 4,
    "2": 2,
    "20": 1,
    "21": 0,
    "22": 3,
    "23": 4,
    "3": 1,
    "4": 1,
    "5": 3,
    "6": 4,
    "7": 1,
    "8": 0,
    "9": 4,
  },
  Object {
    "0": 0,
    "1": 3,
    "10": 7,
    "11": 8,
    "12": 4,
    "13": 6,
    "14": 2,
    "15": 4,
    "16": 6,
    "17": 5,
    "18": 2,
    "19": 1,
    "2": 0,
    "20": 2,
    "21": 0,
    "22": 1,
    "23": 2,
    "3": 3,
    "4": 1,
    "5": 1,
    "6": 2,
    "7": 4,
    "8": 1,
    "9": 8,
  },
  Object {
    "0": 1,
    "1": 1,
    "10": 2,
    "11": 12,
    "12": 4,
    "13": 4,
    "14": 4,
    "15": 4,
    "16": 5,
    "17": 2,
    "18": 1,
    "19": 0,
    "2": 4,
    "20": 0,
    "21": 3,
    "22": 0,
    "23": 2,
    "3": 4,
    "4": 2,
    "5": 4,
    "6": 2,
    "7": 4,
    "8": 6,
    "9": 5,
  },
  Object {
    "0": 3,
    "1": 3,
    "10": 6,
    "11": 7,
    "12": 2,
    "13": 4,
    "14": 3,
    "15": 7,
    "16": 3,
    "17": 4,
    "18": 4,
    "19": 3,
    "2": 1,
    "20": 1,
    "21": 1,
    "22": 0,
    "23": 0,
    "3": 0,
    "4": 3,
    "5": 1,
    "6": 3,
    "7": 5,
    "8": 9,
    "9": 7,
  },
  Object {
    "0": 0,
    "1": 0,
    "10": 3,
    "11": 10,
    "12": 6,
    "13": 6,
    "14": 3,
    "15": 2,
    "16": 6,
    "17": 2,
    "18": 3,
    "19": 1,
    "2": 2,
    "20": 1,
    "21": 4,
    "22": 3,
    "23": 1,
    "3": 2,
    "4": 3,
    "5": 3,
    "6": 5,
    "7": 0,
    "8": 7,
    "9": 4,
  },
  Object {
    "0": 1,
    "1": 3,
    "10": 9,
    "11": 7,
    "12": 2,
    "13": 0,
    "14": 5,
    "15": 4,
    "16": 4,
    "17": 4,
    "18": 2,
    "19": 2,
    "2": 1,
    "20": 3,
    "21": 0,
    "22": 1,
    "23": 2,
    "3": 2,
    "4": 0,
    "5": 1,
    "6": 5,
    "7": 1,
    "8": 2,
    "9": 4,
  },
  Object {
    "0": 3,
    "1": 0,
    "10": 8,
    "11": 3,
    "12": 4,
    "13": 4,
    "14": 4,
    "15": 4,
    "16": 3,
    "17": 2,
    "18": 3,
    "19": 3,
    "2": 1,
    "20": 2,
    "21": 2,
    "22": 1,
    "23": 0,
    "3": 2,
    "4": 4,
    "5": 2,
    "6": 1,
    "7": 1,
    "8": 4,
    "9": 3,
  },
]
`;
